backend/packages/Upgrade/src/api/repositories/LogRepository.ts

```diff
@@ -289,45 +288,85 @@ export class LogRepository extends Repository<Log> {
     // get experiment repository
     const experimentRepo = getRepository(Experiment);
-    return (
-      experimentRepo
-        .createQueryBuilder('experiment')
-        .innerJoin('experiment.queries', 'queries')
-        .innerJoin('queries.metric', 'metric')
-        .innerJoinAndSelect('metric.logs', 'logs')
-        .innerJoinAndSelect(
-          (qb) => {
-            return qb
-              .subQuery()
-              .select([
-                `"individualEnrollment"."userId" as "userId"`,
-                `"individualEnrollment"."experimentId" as "experimentId"`,
-                `"individualEnrollment"."conditionId" as "conditionId"`,
-              ])
-              .distinct()
-              .from(IndividualEnrollment, 'individualEnrollment');
-          },
-          'individualEnrollment',
-          'experiment.id = "individualEnrollment"."experimentId" AND logs."userId" = "individualEnrollment"."userId"'
-        )
-        .innerJoinAndSelect(
-          (qb) => {
-            return qb
-              .subQuery()
-              .select([`${metricString} as value`, 'logs.id as id'])
-              .from(Log, 'logs');
-          },
-          'extracted',
-          'extracted.id = logs.id'
-        )
-        .where('metric.key = :metric', { metric })
-        .andWhere('experiment.id = :experimentId', { experimentId })
-        .andWhere('queries.id = :queryId', { queryId })
-        .andWhere(`jsonb_typeof(${metricString}) = 'number'`)
-    );
+    if (metricType == 'continuous') {
+      return (
+        experimentRepo
+          .createQueryBuilder('experiment')
+          .innerJoin('experiment.queries', 'queries')
+          .innerJoin('queries.metric', 'metric')
+          .innerJoinAndSelect('metric.logs', 'logs')
+          .innerJoinAndSelect(
+            (qb) => {
+              return qb
+                .subQuery()
+                .select([
+                  `"individualEnrollment"."userId" as "userId"`,
+                  `"individualEnrollment"."experimentId" as "experimentId"`,
+                  `"individualEnrollment"."conditionId" as "conditionId"`,
+                ])
+                .distinct()
+                .from(IndividualEnrollment, 'individualEnrollment');
+            },
+            'individualEnrollment',
+            'experiment.id = "individualEnrollment"."experimentId" AND logs."userId" = "individualEnrollment"."userId"'
+          )
+          .innerJoinAndSelect(
+            (qb) => {
+              return qb
+                .subQuery()
+                .select([`${metricString} as value`, 'logs.id as id'])
+                .from(Log, 'logs');
+            },
+            'extracted',
+            'extracted.id = logs.id'
+          )
+          .where('metric.key = :metric', { metric })
+          .andWhere('experiment.id = :experimentId', { experimentId })
+          .andWhere('queries.id = :queryId', { queryId })
+          .andWhere(`jsonb_typeof(${metricString}) = 'number'`)
```

Comment:
If only last line is different in both the query just use if else statement for the last line

Reply:
Updated in this [commit](https://github.com/CarnegieLearningWeb/UpGrade/pull/399/commits/edb15bc97e92f1f48c59b134e28a777f39bb57a0)